### PR TITLE
Enables PMD for EOorg package

### LINKED ISSUE #3238: pom.xml:340-343: Enable PMD check for...
The puzzle `3160-4ce0c9ff` from #3160 has to be resolved: 

https://github.com/objectionary/eo/blob/ba2a9e3861918f0173da5b3609cefb1a52d7e002/eo-runtime/pom.xml#L340-L343

The puzzle was created by Roman Korostinskiy on 24-Jun-24. 

Estimate: 60 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by c71n93**
Closes #3238

---

## PR-Codex overview
The focus of this PR is to add `// NOPMD` trailing comments to package declarations and fix checkstyle issues in the codebase.

### Detailed summary
- Added `// NOPMD` trailing comments to package declarations in various files.
- Fixed `TrailingCommentCheck` issues in package-info files.
- Updated `PackageNameCheck` in test and main files.
- Removed PMD exclusions in the pom.xml file.
- Suppressed specific PMD warnings in specific classes.

> The following files were skipped due to too many changes: `eo-runtime/src/main/java/EOorg/EOeolang/EOsys/EOos$EOname.java`, `eo-runtime/src/test/java/EOorg/EOeolang/EOfailed.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfs/EOfile$EOsize.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOsys/EOposix.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOtry.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfs/EOfile$EOexists.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfs/EOdir$EOmade$EOmkdir.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfs/EOfile$EOis_directory.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfs/EOfile$EOmoved$EOmove.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfs/EOdir$EOtmpfile$EOtouch.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfs/EOfile$EOtouched$EOtouch.java`, `eo-runtime/src/test/java/EOorg/EOeolang/EOsys/DispatchedUnixSyscallTest.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfs/EOfile$EOdeleted$EOdelete.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfs/EOfile$EOopen$EOprocess_file.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOsys/DispatchedUnixSyscall.java`, `eo-runtime/src/main/java/EOorg/EOeolang/Cages.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOor.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOeq.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOand.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOxor.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOnumber$EOgt.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOnumber$EOdiv.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOright.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOnumber$EOplus.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOconcat.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOnumber$EOtimes.java`, `eo-runtime/src/main/java/EOorg/EOeolang/Heaps.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOcage$EOencaged$EOencage.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOslice.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfs/EOdir$EOwalk.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOmalloc$EOof$EOallocated$EOread.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOmalloc$EOof$EOallocated$EOwrite.java`, `eo-runtime/src/test/java/EOorg/EOeolang/EOmallocTest.java`, `eo-runtime/src/test/java/EOorg/EOeolang/EOsys/EOosTest.java`, `eo-runtime/src/test/java/EOorg/EOeolang/EOnumberTest.java`, `eo-runtime/src/test/java/EOorg/EOeolang/EOsys/EOposixTest.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfs/EOfile$EOopen$EOfile_stream$EOread$EOread_bytes.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfs/EOfile$EOopen$EOfile_stream$EOwrite$EOwritten_bytes.java`, `eo-runtime/src/test/java/EOorg/EOeolang/EOcageTest.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOio/EOconsole$EOwrite$EOwritten_bytes.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOio/EOconsole$EOread$EOread_bytes.java`, `eo-runtime/src/main/java/org/eolang/PhSafe.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOtxt/EOregex$EOcompiled.java`, `eo-runtime/src/test/java/EOorg/EOeolang/HeapsTest.java`, `eo-runtime/src/test/java/EOorg/EOeolang/EOfs/FilesTest.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOstring$EOslice.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfs/Files.java`, `eo-runtime/src/test/java/EOorg/EOeolang/EOtryTest.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOrust.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOtxt/EOsscanf.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOtxt/EOsprintf.java`, `eo-runtime/src/test/java/EOorg/EOeolang/EOio/EOconsoleTest.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOtxt/EOregex$EOpattern$EOmatch$EOmatched_from_index.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOerror.java`, `eo-runtime/src/test/java/EOorg/EOeolang/EOio/EOstdinTest.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by c71n93** [comment]
@maxonfjvipon check this one, please

File: eo-runtime/src/main/java/EOorg/EOeolang/Cages.java [hunk 1]
```diff
@@ -24,8 +24,9 @@
 
 /*
  * @checkstyle PackageNameCheck (4 lines)
+ * @checkstyle TrailingCommentCheck (3 lines)
  */
-package EOorg.EOeolang;
+package EOorg.EOeolang; // NOPMD
```
[the remark this answers is not in the document]
Comment (c71n93) [reply eo-runtime/src/main/java/EOorg/EOeolang/Cages.java:29]: @maxonfjvipon check this: https://pmd.github.io/pmd/pmd_userdocs_suppressing_warnings.html, the second item

File: eo-runtime/src/main/java/EOorg/EOeolang/EOerror.java [hunk 2]
```diff
@@ -55,32 +57,59 @@ public final class EOerror extends PhDefault implements Atom {
...
+
     /**
-     * Make a message from an exception.
-     * @param exp The exception
-     * @return Message
+     * Builds error message from the exception.
+     *
+     * @since 0.40
      */
-    public static String message(final Throwable exp) {
```
[the remark this answers is not in the document]
Comment (c71n93) [reply eo-runtime/src/main/java/EOorg/EOeolang/EOerror.java:67]: @maxonfjvipon public static methods are prohibited)

**Review by maxonfjvipon — APPROVED** [review]

File: eo-runtime/src/test/java/EOorg/EOeolang/EOio/EOconsoleTest.java [hunk 3]
```diff
@@ -47,13 +48,19 @@
  */
 @SuppressWarnings("JTCOP.RuleAllTestsHaveProductionClass")
 final class EOconsoleTest {
+
+    /**
+     * Hello.
+     */
+    private static final String HELLO = "Hello";
```
Comment (yegor256) [inline eo-runtime/src/test/java/EOorg/EOeolang/EOio/EOconsoleTest.java:55]: @c71n93 this is not a good idea in general - to share literals between tests. Can't we use a different string in each test?

File: eo-runtime/src/test/java/EOorg/EOeolang/EOio/EOstdinTest.java [hunk 4]
```diff
@@ -47,11 +48,17 @@
  */
 @SuppressWarnings("JTCOP.RuleAllTestsHaveProductionClass")
 final class EOstdinTest {
+
+    /**
+     * Next line.
+     */
+    private static final String NEXT_LINE = "next-line";
```
Comment (yegor256) [inline eo-runtime/src/test/java/EOorg/EOeolang/EOio/EOstdinTest.java:55]: @c71n93 the same here -- don't share literals between tests, use unique ones for each test

**Review by yegor256 — CHANGES_REQUESTED** [review]
@c71n93 a few minor comments from me

File: eo-runtime/src/test/java/EOorg/EOeolang/EOio/EOstdinTest.java (hunk 4 above)
Comment (c71n93) [reply eo-runtime/src/test/java/EOorg/EOeolang/EOio/EOstdinTest.java:55]: @yegor256 Here "next-line" is the name of EO attribute that is used in tests, so we can't use different strings in this case.

**Comment by c71n93** [comment]
@yegor256 fixed, check please

**Review by yegor256 — APPROVED** [review]

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@maxonfjvipon Great job on the review! You've earned +31 points: +25 base, +6 for reviewing 597 hits-of-code (capped at 30), -10 for having only 5 comments, and +5 for those comments. We didn't deduct points for self-review or low hit count, which is a plus! Your balance is now [+238](https://www.eolang.org/zerocracy/objectionary-vitals.html). Keep up the good work, and remember more comments can boost your points next time! 💻👍

**Comment by 0crat** [comment]
@c71n93 Hey there, awesome job on your contribution! 🎉 You've snagged +18 points this time: +25 for the base reward, but we had to dock -7 because your [hits-of-code](https://www.yegor256.com/2014/11/14/hits-of-code.html) (643) crossed the 200 mark. Don't sweat it though – it's still a solid effort! Remember, we love contributions under 200 HOC, but we're thrilled you're putting in the work. Your balance is now sitting pretty at [+105](https://www.eolang.org/zerocracy/objectionary-vitals.html). Keep those contributions rolling in, and let's aim for that sweet spot of concise, impactful code. [Keep up](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) the great work!

**Comment by 0crat** [comment]
@yegor256 Hello! Thank you for your code review contribution. We've calculated your points based on our team policy: +25 base points, +11 for the 597 hits-of-code reviewed, and +8 for your 8 comments. However, we've adjusted the total to +44 points to align with our maximum reward cap of 80 points. Your updated running balance is now [+186](https://www.eolang.org/zerocracy/objectionary-vitals.html). Keep up the great work and remember that detailed feedback helps improve our code quality! 👍

**Pull request merged**